annotators/user_knowledge_memorizer/server.py

```diff
@@ -451,6 +457,63 @@ def check_and_add_properties(graph, prop_triplets: List[dict], user_id: str) ->
     return properties_to_add_to_kg, properties_already_in_kg
 
 
+def get_knowledge(user_id):
+    user_data = kg_graph.search_for_relationships(id_a=user_id)
+    rels = [triplet["rel"] for triplet in user_data]
+    entity_ids = [triplet["id_b"] for triplet in user_data]
+    relationship_entity_id_pair = list(zip(rels, entity_ids))
+    entity_values = kg_graph.get_properties_of_entities(entity_ids)
+    user_triplets = []
+    for entity in entity_values:
+        for rel, id in relationship_entity_id_pair:
+            if entity["@id"] == id:
+                user_triplets.extend([("User", rel, entity.get("substr", entity.get("Name")))])
+
+    return user_triplets
+
+
+def convert_triplets_to_natural_language(triplets):
+    CHAT_GPT_PORT = os.getenv("CHAT_GPT_PORT")
+    OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")
+    assert OPENAI_API_KEY, logger.error("Error: OpenAI API key is not specified in env")
```

Comment:
you can not use chatgpt only considering port. It could be raised in somewhere else, so you won't be able to access it like that. Utilize it giving url.

Reply:
Done